src/main.ts

```diff
@@ -182,6 +184,72 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
     chunks.push(generateDataLoadersType());
   }
 
+  if (options.outputSchema) {
+    chunks.push(code`
+      export interface ProtoMetadata {
+        fileDescriptor: ${imp('IFileDescriptorProto@protobufjs/ext/descriptor')};
+        references: { [key: string]: any };
```

Comment:
How about instead of a single `references` field, we have `messages: { [key: string]: any`, and then `enums` and `clients`.

Just thinking it might be nice to keep those separate in case someone wanted to, say, loop over the clients or loop over just the messages.

Wdyt?

Reply:
I think that considering that the metadata gets generated for everything, if you want to have the list of just enums, you can use `protoMetadata.fileDescriptor.enumType` and the same for messages and services.

The reason I would like to keep this as a single large object is so that if you are given a `ProtoMetadata` instance, and a string reference of `.simple.Nested.InnerMessage.DeepMessage`, you don't have to check 3 different objects for its existence.